redbot/cogs/economy/economy.py

```diff
@@ -247,6 +248,80 @@ def __init__(self, bot: Red):
                 )
             )
 
+    @_bank.group(name="prune")
+    @check_global_setting_admin()
+    async def _prune(self, ctx):
+        """Prune bank accounts."""
+        pass
+
+    @_prune.command(name="local")
+    @commands.guild_only()
+    @checks.guildowner()
+    async def _local(self, ctx, confirmation: bool = False):
+        """Prune bank accounts for users no longer in the server."""
+        global_bank = await bank.is_global()
+        if global_bank is True:
+            return await ctx.send(_("This command cannot be used with a global bank."))
+
+        if confirmation is False:
+            await ctx.send(
+                _(
+                    "This will delete all bank accounts for users no longer in this server."
+                    "\nIf you're sure, type "
+                    "`{prefix}bank prune local yes`"
+                ).format(prefix=ctx.prefix)
+            )
+        else:
+            await bank.bank_prune(self.bot, guild=ctx.guild)
+            await ctx.send(_("Bank accounts for users no longer in this serve have been deleted."))
+
+    @_prune.command(name="global")
+    @checks.is_owner()
+    async def _global(self, ctx, confirmation: bool = False):
+        """Prune bank accounts for users who no longer share servers with the bot."""
+        global_bank = await bank.is_global()
+        if global_bank is False:
+            return await ctx.send(_("This command cannot be used with a local bank."))
+
+        if confirmation is False:
+            await ctx.send(
+                _(
+                    "This will delete all bank accounts for users "
+                    "who no longer share servers with the bot."
+                    "\nIf you're sure, type `{prefix}bank prune global yes`"
+                ).format(prefix=ctx.prefix)
+            )
+        else:
+            await bank.bank_prune(self.bot)
+            await ctx.send(
+                _(
+                    "Bank accounts for users who "
+                    "no longer share servers with the bot have been pruned."
+                )
+            )
+
+    @_prune.command()
+    async def user(self, ctx, user: Union[discord.Member, RawUserIds], confirmation: bool = False):
+        """Delete the bank account of a specified user."""
+        try:
+            name = user.display_name
+            uid = user.id
+        except AttributeError:
+            name = user
+            uid = user
+
+        if confirmation is False:
+            await ctx.send(
+                _(
+                    "This will delete {name}'s bank account."
+                    "\nIf you're sure, type "
+                    "`{prefix}bank prune user {id} yes`"
+                ).format(prefix=ctx.prefix, id=uid, name=name)
+            )
+        else:
+            await bank.bank_prune(self.bot, guild=ctx.guild, user_id=uid)
```

Comment:
`ctx.guild` might be `None` if command is used in DMs.

Reply:
Will now return an error when trying to run the command via DM if the user has a local bank